ravenpy/extractors/forecasts.py

```diff
@@ -38,15 +46,36 @@ def get_hindcast_day(region_coll: fiona.Collection, date, climate_model="GEPS"):
 
 
 def get_CASPAR_dataset(
-    climate_model: str, date: dt.datetime
+    climate_model: str,
+    date: dt.datetime,
+    thredds: str = THREDDS_URL,
+    directory: str = "dodsC/birdhouse/disk2/caspar/daily/",
 ) -> Tuple[
     xr.Dataset, List[Union[Union[DatetimeIndex, Series, Timestamp, Timestamp], Any]]
 ]:
-    """Return CASPAR dataset."""
+    """Return CASPAR dataset.
+
+    Parameters
+    ----------
+    climate_model : str
+        Type of climate model, for now only "GEPS" is supported.
+    date : dt.datetime
+        The date of the forecast.
+    thredds : str
+        The thredds server url. Default: "https://pavics.ouranos.ca/twitcher/ows/proxy/thredds/"
+    directory : str
+        The directory on the thredds server where the data is stored. Default: "dodsC/birdhouse/disk2/caspar/daily/"
+
+    Returns
+    -------
+    xr.Dataset
+        The forecast dataset.
+    """
 
     if climate_model == "GEPS":
         d = dt.datetime.strftime(date, "%Y%m%d")
-        file_url = f"https://pavics.ouranos.ca/twitcher/ows/proxy/thredds/dodsC/birdhouse/disk2/caspar/daily/GEPS_{d}.nc"
```

Comment:
Oh very nice removing the hardcode !!! This will allow for proper testing of new Thredds upgrade.